File: client/src/interfaces/Event.dto.ts
import { GalleryDto } from './Gallery.dto';
import { CategoryDto } from './Category.dto';
import { UserDto } from './User.dto';

export interface EventDto {
  id: number;
  topic: string | undefined;
  location: string | undefined;
  latitude: number;
  longitude: number;
  description: string | undefined;
  startDate: string | undefined;
  endDate: string | undefined;
  category: CategoryDto;
  statusId: number;
  participants: UserDto[] | undefined;
  gallery: GalleryDto;
}

File: server/goadventures/src/main/java/io/softserve/goadventures/dto/EventDTO.java
package io.softserve.goadventures.dto;

import io.softserve.goadventures.models.Category;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class EventDTO {
    private int id;

    private String topic;

    private String startDate;

    private String endDate;

    private String location;

    private Double latitude;

    private Double longitude;

    private String description;

    private int statusId;

    private String category;

    private GalleryDto gallery;
}